models/config_dump.go

```diff
@@ -0,0 +1,385 @@
+		}
+
+		if len(descriptors) == 0 {
+			descriptors = []string{"ALL"}
+		}
+
+		matches = append(matches, map[string]interface{}{
+			"match":       strings.Join(descriptors, "; "),
+			"destination": getListenerDestination(chain.Filters),
+		})
+	}
+	return matches
+}
+
+func getListenerDestination(filters []kubernetes.EnvoyListenerFilter) string {
+	if len(filters) == 0 {
+		return ""
+	}
+
+	for _, filter := range filters {
+		if filter.Name == "envoy.filters.network.http_connection_manager" {
+			typedConfig := filter.TypedConfig
+			if typedConfig.RouteConfig != nil {
+				if cluster := getMatchAllCluster(typedConfig.RouteConfig); cluster != "" {
+					return cluster
+				}
+				vhosts := []string{}
+				for _, vh := range typedConfig.RouteConfig.VirtualHosts {
+					if describeDomains(vh) == "" {
+						vhosts = append(vhosts, describeRoutes(vh))
+					} else {
+						vhosts = append(vhosts, fmt.Sprintf("%s %s", describeDomains(vh), describeRoutes(vh)))
+					}
+				}
+				return fmt.Sprintf("Inline Route: %s", strings.Join(vhosts, "; "))
+			}
+
+			if typedConfig.Rds != nil && typedConfig.Rds.RouteConfigName != "" {
+				return fmt.Sprintf("Route: %s", typedConfig.Rds.RouteConfigName)
+			}
+
+			return "HTTP"
+		} else if filter.Name == "envoy.filters.network.tcp_proxy" {
+			typedConfig := filter.TypedConfig
+			if strings.Contains(typedConfig.Cluster, "Cluster") {
+				return typedConfig.Cluster
+			} else {
+				return fmt.Sprintf("Cluster: %s", typedConfig.Cluster)
+			}
+		}
+	}
+	return "Non-HTTP/Non-TCP"
+}
+
+func describeDomains(vh kubernetes.VirtualHostFilter) string {
+	domains := vh.Domains
+	if len(domains) == 1 && domains[0] == "*" {
+		return ""
+	}
+	return strings.Join(domains, "/")
+}
+
+func describeRoutes(vh kubernetes.VirtualHostFilter) string {
+	routes := make([]string, 0, len(vh.Routes))
+	for _, route := range vh.Routes {
+		routes = append(routes, matchSummary(route.Match))
+	}
+	return strings.Join(routes, ", ")
+}
+
+func getMatchAllCluster(route *kubernetes.RouteConfig) string {
+	vhs := route.VirtualHosts
+	if len(vhs) != 1 {
+		return ""
+	}
+
+	vh := vhs[0]
+	domains := vh.Domains
+	if !(len(domains) == 1 && domains[0] == "*") {
+		return ""
+	}
+
+	if len(vh.Routes) != 1 {
+		return ""
+	}
+
+	r := vh.Routes[0]
+	if prefix, found := r.Match["prefix"]; found && prefix.(string) != "/" {
+		return ""
+	}
+
+	if r.Route == nil || len(r.Route.Cluster) == 0 {
+		return ""
+	}
+
+	if strings.Contains(r.Route.Cluster, "Cluster") {
+		return r.Route.Cluster
+	}
+
+	return fmt.Sprintf("Cluster: %s", r.Route.Cluster)
+}
+
+func getAppDescriptor(chainMatch *kubernetes.FilterChainMatch) string {
+	plainText := []string{"http/1.0", "http/1.1", "h2c"}
+	istioPlainText := []string{"istio", "istio-http/1.0", "istio-http/1.1", "istio-h2"}
+	httpTLS := []string{"http/1.0", "http/1.1", "h2c", "istio-http/1.0", "istio-http/1.1", "istio-h2"}
+	tcpTLS := []string{"istio-peer-exchange", "istio"}
```

Comment:
Where do all these values come from, are they exhaustive or not?

Reply:
It appears to be that yes, they are exhaustive.
I've seen that in the istioctl code: https://github.com/istio/istio/blob/master/istioctl/pkg/writer/envoy/configdump/listener.go#L175

It is useful to have this map in order to give more human readable info.